community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/ast/convert/plannerQuery/PlannerQueryBuilder.scala

```diff
@@ -29,8 +29,11 @@ case class PlannerQueryBuilder(private val q: PlannerQuery, returns: Seq[IdName]
 
   def withReturns(returns: Seq[IdName]): PlannerQueryBuilder = copy(returns = returns)
 
-  def updateGraph(f: QueryGraph => QueryGraph): PlannerQueryBuilder =
-    copy(q = q.updateTailOrSelf(_.updateGraph(f)))
+  def updateQueryGraph(f: QueryGraph => QueryGraph): PlannerQueryBuilder =
+    copy(q = q.updateTailOrSelf(_.updateQueryGraph(f)))
+
+  def updateUpdateGraph(f: UpdateGraph => UpdateGraph): PlannerQueryBuilder =
```

Comment:
This is a bit trippy... How about calling these `amendQueryGraph` and `amendUpdateGraph` instead?